### PR TITLE
Integrated the feedback form into Redux.

### PR DESCRIPTION
**Pull request by aikewoody**
# Description

_Description of the feature or fix_
_Include link to Figma designs if relevant_
_Include link to Notion ticket if relevant_

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

_Please attach screenshots showing the change if relevant_

## Checklist (for submission)

- [ ] Analytics/tracking events have been added (if needed)

## Checklist (for reviewers)

- [ ] Checked against Figma designs (if relevant)
- [ ] Checked that data has been successfully saved to the backend (if relevant)

## Out of scope and potential follow-up

_Are there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?_

### CODE REVIEW

File: src/components/inputs/CheckboxTextInputList.tsx [hunk 1]
```diff
@@ -35,6 +35,8 @@ export function CheckboxTextInputList(props: IProps) {
           <TouchableOpacity
             accessible
             accessibilityRole="checkbox"
+            // eslint-disable-next-line react/no-array-index-key
```
Comment (bionara) [inline src/components/inputs/CheckboxTextInputList.tsx:38]: worth adding this as a global rule I reckon!

**Review by bionara — APPROVED** [review]
very nice..!

File: src/features/reconsent/screens/ReconsentReconsiderScreen.tsx [hunk 2]
```diff
@@ -1,18 +1,39 @@
...
+
+  function onPressPositive() {
+    props.navigation.push('ReconsentRequestConsent', { secondTime: true });
+  }
+
+  async function onPressNegative() {
+    setLoading(true);
+    try {
+      await generalApiClient.postUserEvent('feedback_reconsent', feedbackData);
+    } catch (_) {}
```
Comment (juliatan) [inline src/features/reconsent/screens/ReconsentReconsiderScreen.tsx:31]: Again, don't know if we should show error message here to get user to try again?

File: src/core/state/reconsent/slice/index.ts [hunk 3]
```diff
@@ -1,24 +1,34 @@
...
+      feedbackData: {
+        ...state.feedbackData,
+        [action.payload.feedbackId]: action.payload.value,
+      },
+    }),
   },
 });
 
-export const { updateDiseasePreferences } = reconsentSlice.actions;
-export const selectReconsent = (state: RootState) => state.reconsent;
```
Comment (juliatan) [inline src/core/state/reconsent/slice/index.ts:23]: Why have you deleted this? I think we still need this on other screens to call `diseasePreferences` state.

**Review by juliatan — COMMENTED** [review]
Hey Aike - please can you revert the deletion in reconsent slice?

File: src/features/reconsent/screens/ReconsentReconsiderScreen.tsx (hunk 2 above)
Comment (aikewoody) [reply src/features/reconsent/screens/ReconsentReconsiderScreen.tsx:31]: We had a talk on Slack about this with Christina and Solveig and the conclusion was that if it fails it shouldn't block the user

File: src/core/state/reconsent/slice/index.ts (hunk 3 above)
Comment (aikewoody) [reply src/core/state/reconsent/slice/index.ts:23]: Where exactly? A global search through the codebase for `selectReconsent` gives 0 results

**Pull request merged**